spec/tools/fix_auth/auth_model_spec.rb

```diff
@@ -71,28 +65,28 @@
       end
 
       it "should upgrade old columns" do
-        subject.fix_passwords(old)
+        subject.fix_passwords(old, options)
         expect(old).to be_password_changed
       end
 
-      it "should not encrypt plaintext columns" do
-        subject.fix_passwords(plain)
-        expect(plain).to_not be_password_changed
+      it "should encrypt plaintext columns" do
+        subject.fix_passwords(plain, options)
+        expect(plain).to be_password_changed
```

Comment:
This is the only one I don't understand.  Was it intentional to not encrypt plaintext password columns?  If so, why?

Reply:
ok, this is from the original commit.

There was a lot of focus around upgrading records that needed to go from v1 to v2, but not looking at every record in the database. It took too long time to scan them. So we only brought back records that have the `v1:{}` in it.

Not I don't know if this was a real concern, it was the perceived concern. There would not be business specs around this tool nor would it be an official tool for another year or so.

The original description of the test is it skips records that did not have an encrypted password in them. I believe this was the focus.

Now, I do seem to recall wanting to skip over "password" fields that were not encrypted. The thought is if it hadn't been encrypted yet, then why start now. Think there may have been automate request fields with a field name containing the word password but they were not setup to be encrypted/decrypted. At this time I don't think automate's password handling was as nailed down as it is now. And I also don't think we had full understanding on which fields contained passwords in our app. Alberto and I maintained parallel lists of fields that contained passwords and those were growing during the development of this tool.

I think going forward with this makes sense, but I do have some concern. Guess we find out when we get there.

Reply:
2 issues I see here.

## Quantity

Currently, some tables have hundreds of thousands of records. The animate tables `miq_requests` and `miq_request_responses` come to mind. But not all of those records have passwords. This limiter is there to only parse a subset of the records.

At first, we were concerned about only converting v1 and v0 keys. Since the amount of time it took was unknown skipping the already converted v1 keys was added to the design. Fast forward to today and we are using this tool to convert from one v2 key to another.

## Schema changes

The tables have changed enough over the years along with the keys in the hashes. There is really no way for us to have a definitive list of columns and hash keys that need to be encrypted. To emphasize this point, the `miq_requests` table has user defined keys. So we have been flexible with the values that we encrypt, taking a queue from the data already in the tables.

In this, we have avoided encrypting data that was left unencrypted. It was safer to do let the user define the needs in the dynamic dialog definitions.

So us automatically encrypting columns that we think should be encrypted, when that knowledge has been hard to obtain seems a little presumptuous. 

An example of an issue is a column like `has_password: true` could potentially get encrypted when we don't want the value encrypted.

## TL;DR

Can we only work on records with encrypted values and not encrypt values that were not encrypted before this changed the encryption key?

Reply:
@kbrock Did we come to a resolution on this?  If I recall, we have an in-voice discussion and we ending up being ok with it, but wanted to make sure.

Reply:
Decided to keep this as is for now.

Reply:
lol. I snoozed and loosed on this one.
I really don't feel we should encrypt unencrypted fields here.
But merging is ok